app/src/org/commcare/views/widgets/RecordingFragment.java

```diff
@@ -177,14 +180,46 @@ private void setupRecorder() {
         recorder.setAudioSource(MediaRecorder.AudioSource.MIC);
         recorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
         recorder.setOutputFile(fileName);
-        recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
+        if (isHeAacEncoderSupported()) {
+            recorder.setAudioEncoder(MediaRecorder.AudioEncoder.HE_AAC);
+        } else {
+            recorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB);
+        }
         try {
             recorder.prepare();
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
+    // Checks whether the device supports High Efficiency AAC (HE-AAC) audio codec
+    private boolean isHeAacEncoderSupported() {
+        int numCodecs = MediaCodecList.getCodecCount();
+
+        for (int i = 0; i < numCodecs; i++) {
+            MediaCodecInfo codecInfo = MediaCodecList.getCodecInfoAt(i);
+
+            if (!codecInfo.isEncoder()) {
+                continue;
+            }
+
+            for (String supportedType : codecInfo.getSupportedTypes()) {
+                if (supportedType.equalsIgnoreCase(MIMETYPE_AUDIO_AAC)) {
+                    MediaCodecInfo.CodecCapabilities cap = codecInfo.getCapabilitiesForType(MIMETYPE_AUDIO_AAC);

```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/a4f898b46fc5d5fb063b0a8b7dadd8fe5076e12c/app/src/org/commcare/views/widgets/RecordingFragment.java#L208"></span>Field requires API level 21 (current min is 16): <code>android.media.MediaFormat#MIMETYPE_AUDIO_AAC</code></td>
    </tr>
  </tbody>
</table>

Reply:
https://github.com/dimagi/commcare-android/pull/2533/commits/de372af8d859383db6612bc630b8f6154790397b